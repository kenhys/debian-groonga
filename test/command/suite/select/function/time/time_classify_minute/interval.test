plugin_register functions/time

table_create Timestamps TABLE_PAT_KEY Time

load --table Timestamps
[
{"_key": "2016-05-05 22:29:59.999999"},
{"_key": "2016-05-05 22:30:00.000000"},
{"_key": "2016-05-05 22:30:00.000001"},
{"_key": "2016-05-05 22:39:59.999999"},
{"_key": "2016-05-05 22:40:00.000000"},
{"_key": "2016-05-05 22:40:00.000001"}
]

select Timestamps \
  --sortby _id \
  --limit -1 \
  --output_columns '_key, time_classify_minute(_key, 10)'
