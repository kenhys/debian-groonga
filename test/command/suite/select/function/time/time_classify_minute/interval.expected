plugin_register functions/time
[[0,0.0,0.0],true]
table_create Timestamps TABLE_PAT_KEY Time
[[0,0.0,0.0],true]
load --table Timestamps
[
{"_key": "2016-05-05 22:29:59.999999"},
{"_key": "2016-05-05 22:30:00.000000"},
{"_key": "2016-05-05 22:30:00.000001"},
{"_key": "2016-05-05 22:39:59.999999"},
{"_key": "2016-05-05 22:40:00.000000"},
{"_key": "2016-05-05 22:40:00.000001"}
]
[[0,0.0,0.0],6]
select Timestamps   --sortby _id   --limit -1   --output_columns '_key, time_classify_minute(_key, 10)'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        6
      ],
      [
        [
          "_key",
          "Time"
        ],
        [
          "time_classify_minute",
          null
        ]
      ],
      [
        1462454999.999999,
        1462454400.0
      ],
      [
        1462455000.0,
        1462455000.0
      ],
      [
        1462455000.000001,
        1462455000.0
      ],
      [
        1462455599.999999,
        1462455000.0
      ],
      [
        1462455600.0,
        1462455600.0
      ],
      [
        1462455600.000001,
        1462455600.0
      ]
    ]
  ]
]
